File: src/main/java/com/trade/securities/adapter/in/web/DerivativesTradingInfoController.java
package com.trade.securities.adapter.in.web;

import com.trade.securities.application.port.in.GetDerivativesTradingInfoUseCase;
import com.trade.securities.domain.DerivativesTradingInfo;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/derivatives")
@RequiredArgsConstructor
public class DerivativesTradingInfoController {

    private final GetDerivativesTradingInfoUseCase getDerivativesTradingInfoUseCase;

    @GetMapping("/trading-info/{marketType}")
    public List<DerivativesTradingInfo> getDerivativesTradingInfo(
            @PathVariable String marketType,
            @RequestParam String basDd) {
        return getDerivativesTradingInfoUseCase.getDerivativesTradingInfo(marketType, basDd);
    }
}

File: src/main/kotlin/com/trade/securities/adapter/in/web/DerivativesTradingInfoController.kt
package com.trade.securities.adapter.`in`.web

import com.trade.securities.application.port.`in`.GetDerivativesTradingInfoUseCase
import com.trade.securities.domain.DerivativesTradingInfo
import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.PathParam
import jakarta.ws.rs.Produces
import jakarta.ws.rs.QueryParam
import jakarta.ws.rs.core.MediaType

@ApplicationScoped
@Path("/api/derivatives")
@Produces(MediaType.APPLICATION_JSON)
class DerivativesTradingInfoController(
    private val getDerivativesTradingInfoUseCase: GetDerivativesTradingInfoUseCase
) {

    @GET
    @Path("/trading-info/{marketType}")
    fun getDerivativesTradingInfo(
        @PathParam("marketType") marketType: String,
        @QueryParam("basDd") basDd: String
    ): List<DerivativesTradingInfo> {
        return getDerivativesTradingInfoUseCase.getDerivativesTradingInfo(marketType, basDd)
    }
}
